plugins/auth/auth_base.js

```diff
@@ -83,12 +83,18 @@ exports.check_user = function (next, connection, credentials, method) {
         return;
     }
 
-    var passwd_ok = function (valid, message) {
+    // valid: (true|false)
+    // opts: ({ message, code }|String)
+    var passwd_ok = function (valid, opts) {
```

Comment:
You should use === over == as it's stricter and we can also ensure that we've got a DSN object too, additionally if we aren't using the DSN module, we shouldn't return any enhanced codes.

I would do it this way:

`````javascript
// valid:  true/false
// message:  optional response string or DSN object
var passwd_ok = function (valid, message) {
    var code, reply;
    if (typeof(message) === 'object' && message.constructor.name === 'DSN') {
        code = message.code;
        reply = message.reply;
    }
    else {
        code = (valid ? 235 : 535);
        reply = (message ? message : (valid ? 'Authentication successful' : 
                                              'Authentication failed'));
    }

    ....
    connection.respond(code, reply, function () ....
`````